### PR TITLE
Crane does not redirect correctly to schema 1 path.

### PR DESCRIPTION
**Pull request by ipanova**
closes #2920
https://pulp.plan.io/issues/2920

### CODE REVIEW

**Comment by pcreech** [comment]
2.2-dev created.  Can you target that instead of master?

File: tests/views/test_path.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def test_valid_repo_name_for_manifest(self):
 
         self.assertEqual(response.status_code, 302)
         self.assertTrue(response.headers['Content-Type'].startswith('text/html'))
-        self.assertTrue('foo/bar/manifests/2' in response.headers['Location'])
+        self.assertTrue('foo/bar/manifests/1' in response.headers['Location'])
```
Comment (mhrivnak) [inline tests/views/test_path.py]: Is this just a lazy way of asserting that the redirect URL ends in ``foo/bar/manifests/1.25.1-musl`` ?

**Comment by ipanova** [comment]
@pcreech it should be 3.0 since it comes with pulp_docker 3.0 and has backwards incompatible changes.
so i guess the branch should be 3.0-dev?

File: crane/views/v2.py [hunk 2]
```diff
@@ -99,6 +100,9 @@ def name_redirect(relative_path):
                 path_component = os.path.join(manifest, '2', identifier)
             else:
                 path_component = os.path.join(manifest, '1', identifier)
+        # this is needed for V2Repos
+        else:
+                path_component = os.path.join(manifest, '1', identifier)
```
Comment (ipanova) [inline crane/views/v2.py]: the indentations is correct

```
if schema2_data or manifest_list_data:
    ......
else:
    ......
```

File: tests/views/test_path.py (hunk 1 above)
Comment (ipanova) [inline tests/views/test_path.py]: yes it is, it is enough here to check if the directory is points to schema 1 2, or lists.
Though this tests is failing here, it does pass on my env, i have no idea why it fails in travis, all seems to be good

**Comment by ipanova** [comment]
@mhrivnak no, i am giving up, i asked vrutkovs, locally all tests passed for him as well.

File: crane/views/v2.py [hunk 3]
```diff
@@ -99,6 +100,9 @@ def name_redirect(relative_path):
                 path_component = os.path.join(manifest, '2', identifier)
             else:
                 path_component = os.path.join(manifest, '1', identifier)
+        # this is needed for V3Repo which do not have schema2 manifests
```
Comment (mhrivnak) [inline crane/views/v2.py]: It looks like the V3Repo does have schema 2 manifests. Can you clarify?

``V3Repo = namedtuple('V3Repo', ['url', 'url_path', 'schema2_data', 'protected'])``

Comment (ipanova) [inline crane/views/v2.py]: i meant when the schema2_data is empty list

**Pull request merged**